### PR TITLE
migrate to qt5

### PR DESCRIPTION
**Pull request by nXecure**
Hi.
First, thank you very much for creating dialogbox.
As Debian is removing all gtk2 libraries, I have been searching for a replacement for gtkdialog. I found your project and I am learning how to create dialogs with your project dialogbox. I realized that qt 4.8 libraries are also being removed from Debian, so I tried and think I have succeeded porting your project to qt5. It was extremely easy as you have organized it very well. I only needed to edit 2 lines and now it compiles using qt5 libraries.

There is still a lot I need to learn and test, but I think these changes should also work for you. If you are interested in migrating to qt5 and testing the project with these libraries, please see if it works for you before accepting this merge request.

I hope I can trouble you in the future if I encounter things I don't understand and need a bit of your support.

Regards.

### CODE REVIEW

**Comment by martynets** [comment]
Hi nXecure!
Many thanks for your interest to my project. I do hope it will help you. For sure you are welcome to trouble me :)
Also I thank you for your help with migration to qt5. With no doubts it is needed as it is already 5 years since end of support of qt4.
But I still see qt4 packages (including development ones) available in the Ubuntu repository. Moreover I was unable to test your trivial changes on my current device with lubuntu 18 installed. I simply was unable to install qtbase5-dev package because of dependencies on GLES/DRM stuff (qt5 depends on lower versions than lubuntu desktop).
So, I really want to keep support of both versions of qt 4 and 5. This is doable. I will add comments to your change and kindly ask you to test this on your side and update your pull request. Proposed changes were tested with qt4 and work well.
Thank you in advance!

File: dialogbox.pro [hunk 1]
```diff
@@ -1,6 +1,6 @@
 TEMPLATE = app
 CONFIG += qt thread release
-QT +=
+QT += widgets
```
Comment (martynets) [inline dialogbox.pro]: Please consider adding modules conditionally as it works for both versions of qt:
```
greaterThan(4, QT_MAJOR_VERSION): QT += widgets
```

File: src/dialogbox.h [hunk 2]
```diff
@@ -23,7 +23,7 @@
 #ifndef DIALOGBOX_H_
 #define DIALOGBOX_H_
 
-#include <QtGui>
+#include <QtWidgets>
```
Comment (martynets) [inline src/dialogbox.h]: Also the next code snippet ensures this builds for both versions of qt:
```
#include <QtGlobal>
#if QT_VERSION >= 0x050000
    #include <QtWidgets>
#else
    #include <QtGui>
#endif
```
Please note `QtGlobal` defines the `QT_VERSION`.

File: README.md [hunk 3]
```diff
@@ -33,12 +33,12 @@ The main purpose of this application is to provide GUI functionality to shell sc
 This is a Qt application and it introduces all the power of Qt to shell scripts from command buttons through stylesheets and animations.
 
 #### Software Requirements
-This application is designed using the Qt 4.8 library and this is the only run-time dependency. It must be portable across systems supported by the Qt but the author tested it in GNU/Linux environment only.
+This application was designed for Qt 4.8, but the new versions will be using the Qt 5 library. It must be portable across systems supported by the Qt libraries, but the author tested it in GNU/Linux environment only.
```
Comment (martynets) [inline README.md]: Please rephrase this sentence to say that both versions are supported now.

File: README.md [hunk 4]
```diff
@@ -33,12 +33,12 @@ The main purpose of this application is to provide GUI functionality to shell sc
...
 
 To compile the application from the source code the following packages must also be installed on a GNU/Linux system - build-time dependencies:
 - g++
 - make
-- libqt4-dev
+- qtbase5-dev
```
Comment (martynets) [inline README.md]: Please point that either of these modules can be used depending on the available version of qt.

File: README.md [hunk 5]
```diff
@@ -62,6 +62,11 @@ make install
 ```
 > Note: the application is installed in `/usr/bin` directory and thus the last one command requires root privileges
 
+If you want to make sure that it compiles using Qt5 libraries, select it as the default.
```
Comment (martynets) [inline README.md]: Do you mean a case when multiple versions are installed? If yes, please clarify.

**Comment by nXecure** [comment]
Thanks for answering back. I will follow your suggestions and proceed to test before updating the pull request. You are right that it is better to keep the compatibility with qt4, so there are more options.
My reason for migrating to qt5 is the same you will experience on your next reinstall of Lubuntu (or other Ubuntu based systems):
https://packages.ubuntu.com/search?suite=all&arch=any&searchon=names&keywords=libqt4-dev
libqt4-dev will no longer be available in the future (not included in Ubuntu 20), so qt4 dependent packages will no longer be able to compile without the required development packages.

I was looking for an equivalent to libqt4-dev for qt5, but the only suggestion I found was using the heavily dependent qtbase5-dev package. If may investigate a bit, but based on this answer:
https://askubuntu.com/questions/508503/whats-the-development-package-for-qt5-in-14-04/577334#577334
qt gui module is related to qtbase5-dev dev package, so I don't see much hope.

I will write back when I make the suggested changes and test them.

Regards.

**Comment by nXecure** [comment]
Thanks for the improvements. Switching from qt 4 to qt 5 while building is only editing 2 files in the Debian folder without touching the original source code.

As you were having trouble with the qtbase5-dev package, I have tried building for Ubuntu Bionic (but haven't tested myself if the install and run).
I have build for both qt4 and qt5. If you can test the qt5 build and see that it installs and works as intended on your Lubuntu 18, then we could consider the changes to be working.
https://gitlab.com/nXecure/my-packages/-/tree/master/dialogbox/bionic

Thanks again for creating this project. I have been enjoying learning while building small demo GUIs, and will soon try creating useful programs with bash + dialogbox.

**Comment by martynets** [comment]
Hi nXecure!
Have tested your binary and it works perfect. Many thanks!
Merged your pull request.
Thank you again for your contribution.
Stay in touch.

Regards,
Andriy

**Pull request merged**